File: java/src/test/java/challenges/aoc2024/day4/model/WordsearchLetterTest.java
package challenges.aoc2024.day4.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class WordsearchLetterTest {

    @Test
    void shouldBuildWordsearchLetterObjectFromLetterAndPosition() {
        // When
        final WordsearchLetter letter = WordsearchLetter.of("k", CoordinatePair.of(2, 3));

        // Then
        assertEquals(ImmutableWordsearchLetter.class, letter.getClass());
        assertEquals("k", letter.getLetter());
        assertEquals(CoordinatePair.of(2, 3), letter.getPostion());
    }
}

File: scala/src/test/scala/challenges/aoc2024/day4/model/WordsearchLetterTest.scala
package challenges.aoc2024.day4.model

import model.CoordinatePair
import org.scalatest.funsuite.AnyFunSuite

class WordsearchLetterTest extends AnyFunSuite {

  test("should build WordsearchLetter object from letter and position") {
    // When
    val letter: WordsearchLetter = WordsearchLetter("k", CoordinatePair(2, 3))

    // Then
    assert(letter.isInstanceOf[WordsearchLetter])
    assert("k" == letter.letter)
    assert(CoordinatePair(2, 3) == letter.position)
  }
}
